stories/Who_was_protected_by_the_first_COVID-19_lockdown.ipynb

```diff
@@ -0,0 +1,2412 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Derive Mean IMD Decile\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "At this point we want to join the two datasets together in order to explore the relationship between our two key metrics. \n",
+    "\n",
+    "A problem is that the index of multiple deprivation comes with a geographical granularity at the LSOA level, whilst the COVID-19 mortality counts come with a geographical granularity at the Local Authority District (LAD) level. To complicate things, for each LAD there are generally multiple LSOAs, each with different indexes of multiple deprivation. For more information about the different geographical regions in the UK, read [this](https://data.gov.uk/dataset/ec39697d-e7f4-4419-a146-0b9c9c15ee06/output-area-to-lsoa-to-msoa-to-local-authority-district-december-2017-lookup-with-area-classifications-in-great-britain).\n",
+    "\n",
+    "We need to aggregate the LSOAs into LADs by averaging out the indexes of multiple deprivation. First let's write some functions to help us."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2020-10-07T15:26:14.375680Z",
+     "start_time": "2020-10-07T15:26:14.372220Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "def get_mean_IMD_rank(LAD_code):\n",
+    "    # select relevant LSOAs\n",
+    "    LSOAs = imd_df[imd_df['Local Authority District code (2019)'] == LAD_code]\n",
+    "    \n",
+    "    # calculate mean IMD rank\n",
+    "    mean_IMD_rank = round(LSOAs['Index of Multiple Deprivation (IMD) Rank'].mean(), 2)\n",
+    "    std_IMD_rank = round(LSOAs['Index of Multiple Deprivation (IMD) Rank'].std(), 2)\n",
+    " \n",
+    "    return mean_IMD_rank, std_IMD_rank\n",
+    "\n",
+    "def get_mean_IMD_decile(LAD_code):\n",
+    "    # select relevant LSOAs\n",
+    "    LSOAs = imd_df[imd_df['Local Authority District code (2019)'] == LAD_code]\n",
+    "    \n",
+    "    # calculate mean IMD rank\n",
+    "    mean_IMD_decile = round(LSOAs['Index of Multiple Deprivation (IMD) Decile'].mean(), 2)\n",
+    "    std_IMD_decile = round(LSOAs['Index of Multiple Deprivation (IMD) Decile'].std(), 2)\n",
+    "    \n",
+    "    return mean_IMD_decile, std_IMD_decile"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we can use these functions to calculate the mean IMD decile in each Local Authority District."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2020-10-07T15:26:15.253845Z",
+     "start_time": "2020-10-07T15:26:14.379029Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "LAD_codes = imd_df['Local Authority District code (2019)'].unique()\n",
+    "#mean_IMD_rank, std_IMD_rank = np.vectorize(get_mean_IMD_rank)(LAD_codes)\n",
```

Comment:
These commented out lines could be removed if the IMD ranks aren't needed any longer.